### DISCUSSION TITLE
Can we remove built-in themes from the admin site?

### DISCUSSION THREAD

**Discussion by fuzl-llc** [OrchardCMS/OrchardCore#12210] [Q&A] [answered]
It appears that OrchardCore.Application.Cms.Targets includes the following themes:

1. SafeMode
2. TheAgencyTheme
3. TheBlogTheme
4. TheComingSoonTheme
5. TheTheme

I have my own themes and never expect to use these yet they clutter up the "Admin > Design > Themes" page. Is there any way to remove them and/or hide them from the admin site?

**Comment by jtkech** [comment]
Don't reference them from your web app

**Comment by ns8482e** [comment, marked as answer]
Use https://www.nuget.org/packages/OrchardCore.Application.Cms.Core.Targets Instead

**Comment by jtkech** [comment]
Oops I thought he don't want to see his own themes for now ;)

**Comment by fuzl-llc** [comment]
Awesome, thanks, using the "core" version works! I was worried about the difference and if I'd be removing some other feature I might want but it seems it is just the themes and some translations. For others who may be interested, if you expand OrchardCore.Application.Cms.Targets under the Dependencies node in the VS editor, it includes OrchardCore.Application.Cms.Core.Targets as well as the five themes and OrchardCore.Translations.All. Switching to OrchardCore.Application.Cms.Core.Targets solves the problem above and stops a ton of translations from showing up that I didn't need. That was actually the next problem I was going to work on, getting rid of the Localization folder with all its subfolders and .po files (there's 1,559 Files, 19 Folders, and it takes 27.7 MB). I really didn't want to check that into version control knowing I wasn't going to use it. While using the core targets doesn't ger rid of the Localization folder itself, it does stop all the subfolders and files from showing up. Thanks again!

**Reply by jtkech** [reply]
Yes this is `OrchardCore.Application.Cms.Core.Targets` that embeds the msbuild script `OrchardCore.Application.Cms.Core.Targets.targets` that copies the files from translations packages.

One solution would be to create your own meta package/project with another `.targets` msbuild script, your own `ServiceExtensions.cs` defining your own `AddMyOrchardCms`.

A simpler solution would be to add this in your web app project file.

    <Target Name="ClearTranslationFiles"
      Condition="'$(IsCrossTargetingBuild)' != 'true' and ('$(IsCrossTargetedBuild)' != 'true' or '$(TargetFrameworks' == '' or '$(TargetFrameworks.IndexOf($(TargetFramework)))' == '0')"
      BeforeTargets="CopyPackageTranslationFiles">

      <Message Text="Clearing translation files: $(MSBuildProjectName)" Importance="high" />

      <ItemGroup>
        <PackageTranslationFiles Remove="@(PackageTranslationFiles)" />
      </ItemGroup>

    </Target>

You will still see the message `Copying translation files ...` but normally nothing will get copied.

Note: You can remove the more complex `Condition="...` line but you will get twice the message "Clearing translation files: ..." but that's okay.
